VMBackup/main/Utils/EventLoggerUtil.py

```diff
@@ -0,0 +1,256 @@
+import os
+import threading
+import json
+import sys
+import datetime
+import time
+import uuid
+if sys.version_info[0] == 2:
+    import Queue as queue
+else:
+    # if python version is > 3
+    import queue
+import shutil
+from Utils.LogHelper import FileHelpers,LoggingConstants,Severity
+from Utils.StringHelper import StringHelper
+from Utils.Event import Event
+
+class EventLogger:
+    _instance = None
+    _lock = threading.Lock()
+    
+    
+    def __init__(self, event_directory, severity_level):
+        self.temporary_directory = os.path.join(event_directory, 'Temp')
+        self.space_available_in_event_directory = 0
+        self.event_processing_interval = 0
+        self.disposed = False
+        self.event_processing_task = None  
+        self.current_message_len = 0
+        self.event_logging_enabled = False
+        self.event_logging_error_count = 0
+        self.events_folder = event_directory
+        self.event_logging_enabled = bool(self.events_folder)
+        self.filehelper = FileHelpers()
+
+        if self.event_logging_enabled:
+            self.extension_version = os.path.basename(os.getcwd())
+            self.operation_id = uuid.UUID(int=0)
+            self.log_severity_level = severity_level
+            print("Information: EventLogging severity level setting is ",self.log_severity_level)
```

Comment:
let's add a try-catch block for the if logic

Reply:
Any exception here would be covered when GetInstance Method is called, so avoided it.